Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -133,13 +133,19 @@ class TopSitesPanel: UIViewController {
 
     private func deleteHistoryTileForSite(site: Site, atIndexPath indexPath: NSIndexPath) {
         profile.history.removeSiteFromTopSites(site) >>== {
-            self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount).uponQueue(dispatch_get_main_queue(), block: { result in
-                self.updateDataSourceWithSites(result)
-                self.deleteOrUpdateSites(result, indexPath: indexPath)
-            })
+            self.invalidateTopSites() >>== {
+                self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount).uponQueue(dispatch_get_main_queue(), block: { result in
+                    self.deleteOrUpdateSites(result, indexPath: indexPath)
+                })
+            }
         }
     }
 
+    private func invalidateTopSites() -> Success {
+        profile.history.setTopSitesNeedsInvalidation()
+        return profile.history.invalidateTopSitesIfNeeded() >>> { succeed() }
```

Comment:
Nit: `>>> { succeed() }` :arrow_right: `>>> succeed`.

Substantive: can we avoid the two pref writes by defining:

```
… SQLiteHistory {
    public func refreshTopSitesCache() -> Success {
        let cacheSize = Int(prefs.intForKey(PrefsKeys.KeyTopSitesCacheSize) ?? 0)
        return self.clearTopSitesCache()
            >>> { self.updateTopSitesCacheWithLimit(cacheSize) }
    }
```

and using that both here and inside `invalidateTopSitesIfNeeded`?